patroni/api.py

```diff
@@ -750,6 +753,27 @@ def shutdown_request(self, request):
             _, request = request  # SSLSocket
         return super(RestApiServer, self).shutdown_request(request)
 
+    def get_certificate_serial_number(self):
+        from cryptography import x509, hazmat
+        try:
+            with open(self.__ssl_options['certfile'], "rb") as f:
+                crt = x509.load_pem_x509_certificate(f.read(), hazmat.backends.default_backend())
+                return crt.serial_number
+        except EnvironmentError:
+            logger.exception("Failed to get serial number from certificate %s", self.__ssl_options['certfile'])
+
+    def reload_local_certificate(self):
+        if self.__protocol == 'https':
+            on_disk_cert_serial_number = self.get_certificate_serial_number()
+            if on_disk_cert_serial_number != self.__ssl_serial_number:
+                self._received_new_cert = True
+                self.__ssl_serial_number = on_disk_cert_serial_number
+                return True
+            else:
+                return False
+        else:
+            return False
```

Comment:
These else branches are redundant...
By default None is returned, which is treated as not True.